### PR TITLE
Add Tilt setup

### PR DESCRIPTION
**Pull request by maelk**
This commit adds a setup to deploy BMO in a Tilt environment for
easier development. The image is built and reloaded live from
changes in the code. This commit:

- adds Tiltfile
- adds script to install kustomize locally
- adds script to start kind with a registry

### CODE REVIEW

**Comment by maelk** [comment]
Tilt provides nice features such as automatic reloading. the executable is automatically recompiled when a change is made in the code. It also permits development within a K8S cluster, with the actual deployment files, closer to how it is actually deployed.

**Comment by maelk** [comment]
@kashifest does this lgty ?

File: Tiltfile [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+            provider_details = [provider_details]
+        for item in provider_details:
+            provider_name = item["name"]
+            provider_config = item["config"]
+            if "context" in provider_config:
+                provider_config["context"] = repo + "/" + provider_config["context"]
+            else:
+                provider_config["context"] = repo
+            if "kustomize_config" not in provider_config:
+                provider_config["kustomize_config"] = True
```
Comment (kashifest) [inline Tiltfile:50]: Having a bit difficulty understanding this if `kustomize_config` is not in `provider config` so setting it to true will do what?

**Review by kashifest — COMMENTED** [review]
Have a question but it looks good. 
/lgtm

**Pull request merged**